net_transport.go

```diff
@@ -310,7 +318,7 @@ func (n *NetworkTransport) getProviderAddressOrFallback(id ServerID, target Serv
 	if n.serverAddressProvider != nil {
 		serverAddressOverride, err := n.serverAddressProvider.ServerAddr(id)
 		if err != nil {
-			n.logger.Printf("[WARN] raft: Unable to get address for server id %v, using fallback address %v: %v", id, target, err)
+			n.logger.Warn(fmt.Sprintf("unable to get address for server id %s, using fallback address %v", id, target), err)
```

Comment:
Could we use `hclog.Fmt()` instead of `fmt.Sprintf` here? 
Just something I thought of while looking this over. :)

Reply:
I've redone all of the logging that had formatting and either rephrased it to better fit with the hclog key/value format and/or added hclog.Fmt() instead of Sprintf.